### PR TITLE
new: [dev] add extended misp event

### PR DESCRIPTION
**Pull request by VVX7**
new: [dev] add extended misp event flag to `get_event()`
new: [dev] add extend event as `extend_event()`

### CODE REVIEW

**Comment by Rafiot** [comment]
@iglocska just a question: can we POST `extended` the same way we POST `deleted` in that case? If yes, I'd rather do that as it will make the code of this method more readable.

**Comment by mokaddem** [comment]
> @iglocska just a question: can we POST `extended` the same way we POST `deleted` in that case? If yes, I'd rather do that as it will make the code of this method more readable.

- You can add the `extends_uuid` (works for both id and uuid) parameter when posting on `/events/add`
- In `/events/add/extends:5`, the `extends` keyword is only used in the UI to prefill the extend `Extends Event` field. If you want to extends an event, you have to put and id/uuid in the `extends_uuid` parameter

So I don't think that line will work
https://github.com/MISP/PyMISP/blob/347950fc68449a2a3a85e8a920558bf7ff73fb97/pymisp/api.py#L251

**Comment by Rafiot** [comment]
> > @iglocska just a question: can we POST `extended` the same way we POST `deleted` in that case? If yes, I'd rather do that as it will make the code of this method more readable.
> 
>     * You can add the `extends_uuid` (works for both id and uuid) parameter when posting on `/events/add`

Yeah, when you create an event, `extends_uuid` is definitely possible, `extend_event()` is an helper that should rely on that instead anyway.

>     * In `/events/add/extends:5`, the `extends` keyword is only used in the UI to prefill the extend `Extends Event` field. If you want to extends an event, you have to put and id/uuid in the `extends_uuid` parameter

Clearly, but the proposed method is doing the POST on `'events/view/{event_id}/extended:{event_id}'` not on `'events/add/{event_id}/extended:{event_id}'`

Which is also incorrect anyway, as I think of if.

Anyway, my question was can I do POST of `{'extended': True}` on `'events/view/{event_id}'`, and by looking here, the answer is yes, I think: https://github.com/MISP/MISP/blob/2.4/app/Controller/EventsController.php#L1089

> 
> 
> So I don't think that line will work
> 
> https://github.com/MISP/PyMISP/blob/347950fc68449a2a3a85e8a920558bf7ff73fb97/pymisp/api.py#L251

@VVX7 can you confirm if it works? And if it does, I'd appreciate a test case in `tests/testlive_comprehensive.py`

**Comment by mokaddem** [comment]
> Anyway, my question was can I do POST of {'extended': True} on 'events/view/{event_id}', and by looking here, the answer is yes, I think: https://github.com/MISP/MISP/blob/2.4/app/Controller/EventsController.php#L1089

 Now you can https://github.com/MISP/MISP/commit/ffad8d9189a18e77b7e8498276b42f374bfb4a07 :)

**Comment by Rafiot** [comment]
Thanks! I added the `extended` functionality.

**Comment by VVX7** [comment]
@Rafiot Confirming that `r = self._prepare_request('POST', f'events/add/extends/{eid}', data=event) ` works.  I've added a basic test to check that the event extension is working too.

File: pymisp/api.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def get_event(self, event: Union[MISPEvent, int, str, UUID],
         if deleted:
             data['deleted'] = deleted
         if extended:
-            data['extended'] = deleted
+            data['extended'] = extended
```
Comment (Rafiot) [inline pymisp/api.py:213]: woops... Thank you for that!

**Comment by Rafiot** [comment]
But as I was expecting, your test doesn't work: if you run `tests/testlive_comprehensive.py` against a live instance, you get `AttributeError: 'MISPEvent' object has no attribute 'extensionEvents'`.

And the event isn't extended.

**Comment by Rafiot** [comment]
This is how you can extend an event: 
https://github.com/MISP/PyMISP/blob/2b5e4539372bc4f6a4712ce43c8c42605f530774/tests/testlive_comprehensive.py#L828

**Pull request merged**